### PR TITLE
Feat(configuration) : add setup command

### PR DESCRIPTION
**Pull request by tundeaoni**
Adds setup command as described in this issue https://github.com/kosyfrances/rundeck-zabbix/issues/19

### CODE REVIEW

File: cli/cmd/setup.go [hunk 1]
```diff
@@ -17,21 +17,52 @@ package cmd
...
+			UserName: zabbixUser,
+			Password: zabbixPassword,
+		}
+
+		rundeckConfig := lib.RundeckConfig{
+			Url:    rundeckUrl,
+			ApiKey: rundeckApiKey,
+		}
+
+		newConfig := lib.Config{
```
Comment (osule) [inline cli/cmd/setup.go]: How about naming this `coreConfig` or `rootConfig` as Zabbix and Rundeck?

File: cli/cmd/root.go [hunk 2]
```diff
@@ -18,7 +18,8 @@ import (
 	"fmt"
 	"os"
 
-	homedir "github.com/mitchellh/go-homedir"
+	"github.com/kosyfrances/rundeck-zabbix/lib"
+	"github.com/prometheus/common/log"
```
Comment (osule) [inline cli/cmd/root.go]: Nice package!

File: cli/cmd/setup.go [hunk 3]
```diff
@@ -17,21 +17,52 @@ package cmd
...
 	Use:   "setup",
-	Short: "A brief description of your command",
-	Long: `A longer description that spans multiple lines and likely contains examples
-and usage of using your command. For example:
-
-Cobra is a CLI library for Go that empowers applications.
-This application is a tool to generate the needed files
-to quickly create a Cobra application.`,
+	Short: "initialise tool configuration",
+	Long:  `initialise tool configuration`,
```
Comment (osule) [inline cli/cmd/setup.go]: Since `Long and `Short` text are the same, what do you think about leaving out `Long`?

File: cli/cmd/setup.go [hunk 4]
```diff
@@ -17,21 +17,52 @@ package cmd
...
+			Url:    rundeckUrl,
+			ApiKey: rundeckApiKey,
+		}
+
+		newConfig := lib.Config{
+			Zabbix:  zabbixConfig,
+			Rundeck: rundeckConfig,
+		}
+
+		newConfig.Save()
```
Comment (osule) [inline cli/cmd/setup.go]: I think this should be handled for error.

**Comment by kosyfrances** [comment]
I will be merging this PR, as I need this branch to work with. @tundeaoni will address the comments since they are non-blocking.

File: cli/cmd/setup.go (hunk 4 above)
Comment (tundeaoni) [reply cli/cmd/setup.go]: @osule an exception is thrown on failure with the appropriate exit code

File: cli/cmd/setup.go (hunk 1 above)
Comment (tundeaoni) [reply cli/cmd/setup.go]: It is actually the only config with other variables holding subsets. Do you find it unclear as is right?

File: cli/cmd/root.go (hunk 2 above)
Comment (tundeaoni) [reply cli/cmd/root.go]: thanks

File: cli/cmd/setup.go (hunk 3 above)
Comment (tundeaoni) [reply cli/cmd/setup.go]: I will update this on a new pr

File: cli/cmd/setup.go (hunk 1 above)
Comment (osule) [reply cli/cmd/setup.go]: Yes. However, I do understand the intent. Thanks for clarifying that.

**Pull request merged**